src/odemis/driver/test/xt_client_test.py

```diff
@@ -381,11 +389,11 @@ def test_set_scan_field_size(self):
         init_scan_size = self.microscope.get_scanning_size()[0]
         new_scanfield_x = scanfield_range['x'][1]
         self.microscope.set_scanning_size(new_scanfield_x)
-        self.assertEqual(self.microscope.get_scanning_size()[0], new_scanfield_x)
+        self.assertAlmostEqual(self.microscope.get_scanning_size()[0], new_scanfield_x)
         # Test it still works for different values.
         new_scanfield_x = scanfield_range['x'][0]
         self.microscope.set_scanning_size(new_scanfield_x)
-        self.assertEqual(self.microscope.get_scanning_size()[0], new_scanfield_x)
+        self.assertAlmostEqual(self.microscope.get_scanning_size()[0], new_scanfield_x)
```

Comment:
By default assertAlmostEqual rounds to 7 decimal places. The minimum range for the scanfield is 1e-8, so I think you want to test at least for 8 decimal places.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual

Reply:
Good catch. Indeed, with microscopes, small is sometimes still very big!
Our own method almost_equal uses 1e-18, which is a little safer.